### PR TITLE
[DNM] [rancher-monitoring] Fix Grafana links in local cluster

### PR DESCRIPTION
**Pull request by stormqueen1990**
## Problem

When some links are used to navigate around the Grafana dashboard, the UI breaks with an HTTP 404 error. After some investigation, it appears the UI is being served under `/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy/`, but our `nginx.conf` rules expect it to be `/k8s/clusters/local/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy/`.

I have not been able to figure out where the redirection from `/k8s/clusters/local/api/v1` to `/api/v1` comes from.

## Solution

Add a condition to serve Grafana from `/api/v1` in the local cluster, and from `/k8s/clusters/<cluster-id>/api/v1` on downstream clusters.

### Manual Testing

1. Install the Rancher Monitoring chart in a local cluster.
2. Navigate to the **Monitoring** tab.
3. Click **Grafana**.
4. Check the UI loads without showing an HTTP 404 cluster.
5. Repeat steps in a downstream cluster.

### Automated Testing

This chart does not have Hull testing yet.

### CODE REVIEW

**Comment by sebracs** [comment]
Shouldn't the rewrite also be fixed with an if/else? See my example here: https://github.com/rancher/rancher/issues/40873#issuecomment-1487616360

File: charts/rancher-monitoring/102.1.0-rc1+up40.1.2/charts/grafana/templates/nginx-config.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          {{- else -}}
+          sub_filter '"appSubUrl":""' '"appSubUrl":"/k8s/clusters/{{ .Values.global.cattle.clusterId }}/api/v1/namespaces/{{ template "grafana.namespace" . }}/services/http:{{ template "grafana.fullname" . }}:{{ .Values.service.port }}/proxy"';
+          {{- end -}}
+          sub_filter ':"/avatar/' ':"avatar/';
+
+          if ($request_filename ~ .*\.(?:js|css|jpg|jpeg|gif|png|ico|cur|gz|svg|svgz|mp4|ogg|ogv|webm)$) {
+            expires             90d;
+          }
+
+          rewrite ^/k8s/clusters/.*/proxy(.*) /$1 break;
```
Comment (sebracs) [inline charts/rancher-monitoring/102.1.0-rc1+up40.1.2/charts/grafana/templates/nginx-config.yaml:88]: Shouldn't the rewrite also be fixed with an if/else? See my example here: https://github.com/rancher/rancher/issues/40873#issuecomment-1487616360

Comment (stormqueen1990) [reply charts/rancher-monitoring/102.1.0-rc1+up40.1.2/charts/grafana/templates/nginx-config.yaml:88]: When I tried this updated version it didn't appear to be necessary.

**Comment by stormqueen1990** [comment]
Superseded by #2596.

File: charts/rancher-monitoring/102.1.0-rc1+up40.1.2/charts/grafana/templates/nginx-config.yaml (hunk 1 above)
Comment (kOstovic) [reply charts/rancher-monitoring/102.1.0-rc1+up40.1.2/charts/grafana/templates/nginx-config.yaml:88]: it appears to be necessary

**Pull request closed without merging**